host/volume/manager/manager.go

```diff
@@ -30,20 +30,30 @@ type Manager struct {
 	namedVolumes map[string]string
 }
 
-func New(p volume.Provider) *Manager {
-	return &Manager{
-		defaultProvider: p,
-		providers:       map[string]volume.Provider{"default": p},
-		volumes:         map[string]volume.Volume{},
-		namedVolumes:    map[string]string{},
+func New(defProvFn func() (volume.Provider, error)) (*Manager, error) {
+	m := &Manager{
+		providers:    map[string]volume.Provider{},
+		volumes:      map[string]volume.Volume{},
+		namedVolumes: map[string]string{},
+	}
+	if _, ok := m.providers["default"]; !ok {
+		p, err := defProvFn()
```

Comment:
Correct; this is one of the parts that makes more sense with the next step in #924.  See https://github.com/flynn/flynn/pull/924/files#diff-dbf2e862dcaac249b6826c1bb5263118R91